src/Google.Api.Gax/ApiCallRetryExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Google.Protobuf;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    public static class ApiCallRetryExtensions
+    {
+        public static ApiCall<TRequest, TResponse> WithRetry<TRequest, TResponse>(
+            this ApiCall<TRequest, TResponse> originalCall,
+            RetrySettings retrySettings,
+            IClock clock,
+            IScheduler scheduler)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            if (retrySettings == null)
```

Comment:
This makes the precondition a few lines later a bit odd. Do we ever actually want to call this without a retry settings?